lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -123,6 +123,18 @@ class SPIRVRegularizeLLVMBase {
   void expandVEDWithSYCLTypeSRetArg(Function *F);
   void expandVIDWithSYCLTypeByValComp(Function *F);
 
+  // According to the specification, the operands of a shift instruction must be
+  // a scalar/vector of integer. When LLVM-IR contains a shift instruction with
+  // i1 operands, they are treated as a bool. We need to extend them to i32 to
+  // comply with the specification. For example:
+  // %21 = load i1, i1 addrspace(4)* %20
+  // %22 = load i32, i32 addrspace(4)* %9
+  // %23 = trunc i32 %22 to i1
+  // %24 = lshr i1 %21, %23
+  // %25 = zext i1 %24 to i32
+  // %24 should be changed to take (i32(%21), %22); %25 users should use %24
```

Comment:
I believe there is no need for such a detailed IR. Let's simplify this in favor of readability, e.g.:
```suggestion
  // %shift = lshr i1 0, 1
  // The instruction should be changed to the extended version "%shift = i32 0, 1" so the args are treated as int operands;
```